### PR TITLE
Feature/add test framework

### PR DESCRIPTION
**Pull request by jurrien-fakkeldij**
Added feature test framework (cucumber/godog)
Written first feature test

### CODE REVIEW

File: features/steps/command_steps.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+func (s *CommandSteps) theResponseShouldBeEphimeral() error {
+	if s.MockSession.Response.Data.Flags != discordgo.MessageFlagsEphemeral {
+		return fmt.Errorf("Response is not ephemiral")
+	}
+	return nil
+}
+
+func (s *CommandSteps) InitializeScenario(ctx *godog.ScenarioContext) error {
```
Comment (koenbollen) [inline features/steps/command_steps.go:78]: I'd put this on the top of this file.

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,15 @@ go 1.22.1
 
 require (
 	github.com/bwmarrin/discordgo v0.27.1 // indirect
+	github.com/cucumber/gherkin/go/v26 v26.2.0 // indirect
+	github.com/cucumber/godog v0.14.0 // indirect
```
Comment (koenbollen) [inline go.mod:8]: did you run `go mod tidy` before committing? I expect this to be a direct dependency

File: features/player_commands.feature [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Feature: Player Commands
+	Scenario: non existant player registers with name
+		When any user sends a "register_player" command with "test_user" name as a parameter
+		Then a response should be given
+		And the response should be "You have registered yourself with the name test_user"
```
Comment (koenbollen) [inline features/player_commands.feature:5]: This could be one step imho

**Review by koenbollen — COMMENTED** [review]
Nice! Looks great.

Some notes:
- Try to not have as many formatting fixes in a PR that also changes logics (e.g. make one PR that fixes the formatting of the entire project).
- Check out these reads on what kind of language you _should_ use in step defs:
  - https://specflow.org/gherkin/gherkin-conventions-for-readable-specifications/
  - https://automationpanda.com/2017/01/30/bdd-101-writing-good-gherkin/
  
Nice work!

File: go.mod (hunk 2 above)
Comment (jurrien-fakkeldij) [reply go.mod:8]: nope, thanks!

**Comment by jurrien-fakkeldij** [comment]
Regarding the formatting, think it was mainly because I moved to a var ( ) with multiple definitions in it. But I'll take it along for next time. 

Thanks for the resources, will have a look at them and check if I want to change some of the naming

**Pull request merged**